src/lib.rs

```diff
@@ -1365,6 +1391,79 @@ impl DnsCache {
     }
 }
 
+pub trait AsIpv4Addrs {
+    fn as_ipv4_addrs(&self) -> Result<HashSet<Ipv4Addr>>;
+}
+
+impl AsIpv4Addrs for &str {
+    fn as_ipv4_addrs(&self) -> Result<HashSet<Ipv4Addr>> {
+        let mut addrs = HashSet::new();
+
+        let iter = self
+            .split(',')
+            .map(str::trim)
+            .map(std::net::Ipv4Addr::from_str);
+
+        for addr in iter {
+            let addr = addr.map_err(|err| Error::ParseIpAddr(err.to_string()))?;
+
+            addrs.insert(Ipv4Addr::from_std(&addr));
+        }
+
+        Ok(addrs)
+    }
+}
+
+impl AsIpv4Addrs for String {
+    fn as_ipv4_addrs(&self) -> Result<HashSet<Ipv4Addr>> {
+        self.as_str().as_ipv4_addrs()
+    }
+}
+
+impl<I: AsIpv4Addrs> AsIpv4Addrs for &[I] {
+    fn as_ipv4_addrs(&self) -> Result<HashSet<Ipv4Addr>> {
+        let mut addrs = HashSet::new();
+
+        for result in self.iter().map(I::as_ipv4_addrs) {
+            addrs.extend(result?);
+        }
+
+        Ok(addrs)
+    }
+}
+
+/// Optimization for zero sized/empty values, as `()` will never take up any space or evaluate to
+/// anything, helpful in contexts where we just want an empty value.
+impl AsIpv4Addrs for () {
+    fn as_ipv4_addrs(&self) -> Result<HashSet<Ipv4Addr>> {
+        Ok(HashSet::new())
+    }
+}
+
+impl<I: AsIpv4Addrs, const N: usize> AsIpv4Addrs for [I; N] {
```

Comment:
const generics is stable after Rust 1.51, but currently we support earliest to Rust 1.49.  How much benefit this support of fixed array bring? 

Since you already support slice `&[I]`,  it seems to me that we don't really need to implement this trait for fixed array directly.  

But if I missed something and there is indeed use case for this,  I will be OK to bump up required Rust version to 1.51.  

Btw, I didn't find how to allow your PR to run GitHub action workflow automatically.  It's probably only a first PR thing, I will run it as soon as I see it.  Sorry about that delay.

Reply:
Thanks for all the feedback. This was more or less a by-product of my experimentation to easily allow an empty value (which was solved with the `()` implementation). You are entirely correct, with `&[T]` already being implemented, `[T; N]` is redundant